### PR TITLE
added roles, user details service and configured authentication

### PR DESCRIPTION
**Pull request by Martynice**

### CODE REVIEW

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -24,6 +29,7 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
+        user.setRoles(Set.of(roleService.getRoleByName("USER")));
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java:32]: Magic words and numbers...

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email);
+        if (user != null) {
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: Do we need this check? ^_^

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            UserBuilder builder = org.springframework.security.core.userdetails
+                    .User.withUsername(user.getEmail());
+            builder.password(user.getPassword());
+            builder.roles(user.getRoles().stream()
+                    .map(x -> x.getRoleName().toString())
+                    .toArray(String[]::new));
+            return builder.build();
+        }
+        throw new UsernameNotFoundException("Couldn't find user by email " + email);
```
Comment (Nokafan) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: Probably you can throw it at the place where you get the user out of optional?

Comment (Sofasmile) [reply src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: we can return optional from service and use orElseThrow here or leave it as it is (without throwing exception)

**Review by Sofasmile — COMMENTED** [review]
travis failed

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java (hunk 3 above)
Comment (Martynice) [reply src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: I check it in service layer

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**